packages/lesswrong/components/review/QuickReviewPage.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+import { useMulti } from '../../lib/crud/withMulti';
+import { getReviewPhase, REVIEW_YEAR, ReviewYear } from '../../lib/reviewUtils';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import sortBy from 'lodash/sortBy';
+import { preferredHeadingCase } from '../../themes/forumTheme';
+import { getVotePower } from '@/lib/voting/vote';
+import { useCurrentUser } from '../common/withUser';
+
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: { 
+    marginBottom: -20
+  },
+  sortingOptions: {
+    whiteSpace: "pre",
+    display: "flex",
+    [theme.breakpoints.down('xs')]: {
+      paddingTop: 12,
+      paddingLeft: 4
+    }
+  },
+  reviewProgressBar: {
+    marginRight: "auto"
+  },
+  postRoot: {
+    position: "relative",
+    paddingLeft: 6,
+    backgroundColor: theme.palette.background.pageActiveAreaBackground,
+    borderBottomWidth: 2,
+    borderBottom: theme.palette.border.itemSeparatorBottom,
+  },
+  loading: {
+    opacity: .5
+  },  
+  loadMore: {
+    ...theme.typography.body2,
+    color: theme.palette.primary.main,
+    marginRight: "auto"
+  }
+});
+
+export const QuickReviewPage = ({classes, reviewYear}: {
+  classes: ClassesType,
+  reviewYear: ReviewYear
+}) => {
+  const [expandedPost, setExpandedPost] = useState<PostsReviewVotingList|null>(null)
+  const [truncatePosts, setTruncatePosts] = useState<boolean>(true)
+  const currentUser = useCurrentUser()
+
+  const { results: posts, loadMore, loading, totalCount } = useMulti({
+    terms: {
+      view: "reviewQuickPage",
+      before: `${reviewYear+1}-01-01`,
+      after: `${reviewYear}-01-01`,
+      limit: 60,
+    },
+    collectionName: "Posts",
+    fragmentName: 'PostsReviewVotingList',
+    fetchPolicy: 'cache-and-network',
+    enableTotal: true,
+    itemsPerPage: 1000,
+    skip: !reviewYear
+  });
+
+  const { PostsItem, SectionFooter, Loading, PostInteractionStripe } = Components
+
+  const sortedPostsResults = !!posts ? [...posts].sort((post1, post2) => {
+    const post1QuadraticScore = post1.currentUserReviewVote?.quadraticScore ?? 0
+    const post2QuadraticScore = post2.currentUserReviewVote?.quadraticScore ?? 0
+    const post1KarmaVote = post1.currentUserVote
+      ? getVotePower({ user: currentUser!, voteType: post1.currentUserVote, document: post1 })
+      : 0;
+    const post2KarmaVote = post2.currentUserVote
+      ? getVotePower({ user: currentUser!, voteType: post2.currentUserVote, document: post2 })
+      : 0;
+    const post1ReadStatus = post1.lastVisitedAt ? 1 : 0
+    const post2ReadStatus = post2.lastVisitedAt ? 1 : 0
+
+    // Sort by karma vote, then quadratic score
+    return post2ReadStatus - post1ReadStatus || post2KarmaVote - post1KarmaVote || post2QuadraticScore - post1QuadraticScore || post2.baseScore - post1.baseScore
```

Comment:
I understand what this is doing but the way it's written makes me sad.  Would prefer early returns for the first two cases:

```
    const post1ReadStatus = post1.lastVisitedAt ? 1 : 0
    const post2ReadStatus = post2.lastVisitedAt ? 1 : 0
    const readStatusDiff = post2ReadStatus - post1ReadStatus;
    if (readStatusDiff !== 0) {
      return readStatusDiff;
    }

...
```